src/hooks/usePreferredLanguage.ts

```diff
@@ -0,0 +1,24 @@
+import { useSyncExternalStore } from "react";
+
+const usePreferredLanguageSubscribe = (
+  callback: (this: Window, event: Event) => any,
+) => {
+  window.addEventListener("languagechange", callback);
+  return () => window.removeEventListener("languagechange", callback);
+};
+
+const getPreferredLanguageSnapshot = () => {
+  return navigator.language;
+};
+
+const getPreferredLanguageServerSnapshot = () => {
+  throw Error("usePreferredLanguage is a client-only hook");
+};
+
+export default function usePreferredLanguage() {
```

Comment:
issue: Could you add the JSDoc (with types) please ?

Reply:
Yeah, how does this sound:
```
/**
 * React hook to get the user's preferred language
 * @returns {string} The user's preferred language
 */
```